src/handlers/actions/emitDomainReputation.ts

```diff
@@ -41,23 +25,35 @@ export default async (event: ContractEvent): Promise<void> => {
     ? ColonyActionType.EmitDomainReputationReward
     : ColonyActionType.EmitDomainReputationPenalty;
 
-  const colonyClient = await getCachedColonyClient(colonyAddress);
+  let domain;
+  let nextToken: string | undefined;
+  /*
+   * Search for domain with skillId from Colony's domains. We paginate the query
+   * just in case there are more domains than the query limit. We can stop searching
+   * if we find a domain.
+   */
+  do {
+    const { data } =
+      (await query<GetColonyQuery, GetColonyQueryVariables>(GetColonyDocument, {
```

Comment:
All I'm doing here is querying the colony, and filtering its domains by the skillId in the event. 

The reason it's inside a loop is in case there are more domains than the query limit (I've set it manually to 1,000), we won't get all the domains unless we paginate the query. Pretty defensive, but there just in case.